src/transformers/models/groupvit/modeling_groupvit.py

```diff
@@ -0,0 +1,1864 @@
+            batch_size,
+            self.num_heads,
+            query_length,
+            key_length,
+        ), f"{attn.shape} vs {(batch_size, self.num_heads, query_length, key_length)}"
+
+        # [batch_size, num_heads, query_length, channels//num_heads] -> [batch_size, query_length, channels]
+        out = (attn @ v).transpose(1, 2).reshape(batch_size, query_length, channels)
+        out = self.proj(out)
+        return out
+
+
+class GroupViTCrossAttentionBlock(nn.Module):
+    def __init__(self, config: GroupViTVisionConfig):
+        super().__init__()
+        self.attn = GroupViTAttention(config)
+        self.norm2 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.mlp = GroupViTMLP(config)
+        self.norm_post = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+
+    def forward(self, query, key):
+        x = query
+        x = x + self.attn(query, key)
+        x = x + self.mlp(self.norm2(x))
+        x = self.norm_post(x)
+        return x
+
+
+class GroupViTAssignAttention(nn.Module):
+    def __init__(self, config: GroupViTVisionConfig):
+        super().__init__()
+        self.scale = config.hidden_size**-0.5
+
+        self.q_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.k_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.v_proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.proj = nn.Linear(config.hidden_size, config.hidden_size)
+        self.assign_eps = config.assign_eps
+
+    def get_attn(self, attn, gumbel=True, hard=True):
+
+        if gumbel and self.training:
+            attn = gumbel_softmax(attn, dim=-2, hard=hard)
+        else:
+            if hard:
+                attn = hard_softmax(attn, dim=-2)
+            else:
+                attn = nn.functional.softmax(attn, dim=-2)
+
+        return attn
+
+    def forward(self, query, key):
+        batch_size, query_length, channels = query.shape
+        value = key
+        key_length = key.size(1)
+        # [batch_size, query_length, channels]
+        q = self.q_proj(query)
+
+        # [batch_size, key_length, channels]
+        k = self.k_proj(key)
+
+        # [batch_size, key_length, channels]
+        v = self.v_proj(value)
+
+        # [batch_size, query_length, key_length]
+        raw_attn = (q @ k.transpose(-2, -1)) * self.scale
+
+        attn = self.get_attn(raw_attn)
+        soft_attn = self.get_attn(raw_attn, gumbel=False, hard=False)
+
+        attn = attn / (attn.sum(dim=-1, keepdim=True) + self.assign_eps)
+
+        assert attn.shape == (batch_size, query_length, key_length)
+
+        out = attn @ v
+
+        out = self.proj(out)
+
+        return out, soft_attn
+
+
+class GroupViTTokenAssign(nn.Module):
+    def __init__(
+        self,
+        config: GroupViTVisionConfig,
+        *,
+        num_group_token,
+        num_output_group,
+    ):
+        super(GroupViTTokenAssign, self).__init__()
```

Comment:
```suggestion
        super().__init__()
```
We don't support Python 2 :-)